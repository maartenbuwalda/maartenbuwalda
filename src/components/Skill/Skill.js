import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import VendorLogo from '../VendorLogo'
import colors from '../../constants/colors'
import LazyLoad from 'react-lazyload'

const StyledSkill = styled.div`
  animation: appear 1s ease;
  background: ${colors.white};
`

const SkillDescription = styled.p`
  display: none;
  margin: 0;
  padding: 2rem;
  text-align: left;
  border-top: 1px solid ${colors.gray};
`

const LogoWrapper = styled.div`
  max-height: 3rem;
  padding: 1rem;
  display: grid;
  grid-template-columns: 3rem 1fr;

  img {
    min-width: 2rem;
  }

  div {
    display: flex;
    align-items: center;
  }

  span {
    text-transform: capitalize;
    font-size: 1.2rem;
    margin-left: 1rem;
  }
`

const Skill = ({ name, children }) => {
  return (
    <LazyLoad
      height={80}
      offset={-100}>
      <StyledSkill>
        <LogoWrapper>
          <VendorLogo vendor={name} />
          <div>
            <span>{name}</span>
          </div>
        </LogoWrapper>
        {children && (
          <SkillDescription>
            {children}
          </SkillDescription>
        )}
      </StyledSkill>
    </LazyLoad>
  )
}

Skill.propTypes = {
  name: PropTypes.string,
  children: PropTypes.node,
}

export default Skill
