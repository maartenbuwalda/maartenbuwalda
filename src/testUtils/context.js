const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
