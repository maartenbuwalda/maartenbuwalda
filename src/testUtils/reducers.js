export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
