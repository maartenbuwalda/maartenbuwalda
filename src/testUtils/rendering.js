import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
