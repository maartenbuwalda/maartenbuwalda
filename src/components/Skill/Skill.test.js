import React from 'react'
import Skill from './index'
import { componentRenders } from '../../testUtils/rendering'

jest.mock('../../constants/images', () => ({}))

describe('Skill component', () => {
  const component = <Skill />
  componentRenders(component)
})
