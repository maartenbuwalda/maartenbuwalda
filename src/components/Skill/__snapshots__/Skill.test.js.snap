// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill component renders correctly 1`] = `
<LazyLoad
  height={80}
  offset={-100}
  once={false}
  overflow={false}
  resize={false}
  scroll={true}
  unmountIfInvisible={false}
>
  <styled.div>
    <styled.div>
      <VendorLogo />
      <div>
        <span />
      </div>
    </styled.div>
  </styled.div>
</LazyLoad>
`;
